flink-table/flink-sql-parser/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1634,16 +1634,52 @@ SqlEndStatementSet SqlEndStatementSet() :
 SqlNode SqlRichExplain() :
 {
     SqlNode stmt;
+    Set<String> explainDetails = new HashSet<String>();
 }
 {
-    <EXPLAIN> [ <PLAN> <FOR> ]
+    <EXPLAIN> 
+    [
+        <PLAN> <FOR> 
+        |
+        (
+            <ESTIMATED_COST> 
+            | 
+            <CHANGELOG_MODE> 
+            | 
+            <JSON_EXECUTION_PLAN>
+        ) {
+            if (explainDetails.contains(token.image.toUpperCase())) {
+                throw SqlUtil.newContextException(getPos(), 
+                ParserResource.RESOURCE.explainDetailRepeat());
+            } else {
+                    explainDetails.add(token.image.toUpperCase());
+            }
+        }
+        (
+            <COMMA>
+            (
+                <ESTIMATED_COST> 
+                | 
+                <CHANGELOG_MODE> 
+                | 
+                <JSON_EXECUTION_PLAN>
+            ) {
+                if (explainDetails.contains(token.image.toUpperCase())) {
+                    throw SqlUtil.newContextException(getPos(), 
+                    ParserResource.RESOURCE.explainDetailRepeat());
+                } else {
+                    explainDetails.add(token.image.toUpperCase());
+                }
+            }
```

Comment:
What about introduce a function like below 

```
void ParseExplainDetail(Set<String> explainDetails) :
{
}
{
    (
        <ESTIMATED_COST>
        |
        <CHANGELOG_MODE>
        |
        <JSON_EXECUTION_PLAN>
    )
    {
        if (explainDetails.contains(token.image.toUpperCase())) {
            throw SqlUtil.newContextException(getPos(),
            ParserResource.RESOURCE.explainDetailRepeat());
        } else {
                explainDetails.add(token.image.toUpperCase());
        }
    }
}
```

In this way we can reuse the codes.